Thumbnail rework

po5 opened the pull request:
Higher quality and lower filesize thumbnails with `cjpeg` (provided by mozjpeg for quality or libjpeg-turbo for speed).

Videos in the upload form now have a preview that matches the final thumbnail.

Thumbnails are now used as placeholder while the full-size image is loading on post pages.  
Thumbnail stays if the full-size image load fails. If both pics fail, the transparency grid is shown (if enabled).

Videos have two thumbnails, one taken at 30% into the video (the "custom" thumbnail shown in search results), and one of the first frame (the thumbnail used as a preroll on post pages).

Images are now resized in RGB even for indexed color PNGs. It would lead to ugly dithering that also hurt compression.
Before 
![image](https://github.com/rr-/szurubooru/assets/42466980/950e07a9-70c8-4718-81b4-7f4f3b527d59)
After
![image](https://github.com/rr-/szurubooru/assets/42466980/bb378e7a-153b-4d69-9fd8-6f3fe72e472c)

Thumbnail filenames are prefixed with `sample_` to make it clear to the user they didn't download the original image.

neobooru commented on client/css/core-general.styl:
```diff
@@ -284,28 +284,25 @@ a .access-key
-        background-size: 20px 20px
-    img
+            repeating-linear-gradient(45deg, $window-color, $window-color 10px, #e6e6e6 10px, #e6e6e6 20px)
+    img, video
```
I assume these changes should be taken over https://github.com/rr-/szurubooru/pull/642/files#diff-965c9ad95631d07c4423bbe276db1d09308e9d790f997c4232f3896b6405b407 right?

po5 commented:
Rebased on top of master. Removed some unrelated changes that I accidentally left in the original PR (stripes instead of transparency grid, contain instead of cover for thumbnails, ruffle for swf thumbnails). Fixed some oversights. Updated tests.

Should be fully working, Docker setup is updated too.
For existing instances, `szuru-admin --regenerate-thumbnails` is required to set up new thumbnail filenames that start with `sample_`, and to give videos separate search and preroll thumbnails.

rr- commented:
>For existing instances, szuru-admin --regenerate-thumbnails is required to set up new thumbnail filenames that start with sample_, and to give videos separate search and preroll thumbnails.

I'd decouple this change from the backend replacement to not let it block it. As a solution to the mentioned problem, instead, I'd try to create a migration that traverses the filesystem and renames the files to the expected pattern.
